### ISSUE TITLE
[Proposal] Optimize dataset loading and activation store

### ISSUE THREAD

**Issue by Hzfinfdu** [OpenMOSS/Language-Model-SAEs#9]
The current activation store implementation has some drawbacks. Maybe we need to add some new features for streaming activation store and make some optimizations. Below I list some details.

1. Text Dataset Collate Config
We need to support SAE training on both pretraining and SFT data, unlike [Anthropic's Scaling Monosemanticity](https://transformer-circuits.pub/2024/scaling-monosemanticity/index.html) in which only pretrained data is used to train SAEs on a supervised finetuned model.

IMO pretraining data should be packed and SFT data should be sorted by length and batched with post paddings. Activations in the residual stream of <PAD>s should be ignored in SAE training. I believe this is better fitted to real-world distribution.

We need to add into the configuration to configure this.

- [x] Support two types of activation gen

2. Shuffle
When training SAEs with data from multiple distributions, shuffling should be an option to add to diversity of information in a batch. This can be implemented by filling in the activation buffer with random sources.

- [x] Support buffer filling from multiple sources
